desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphLinkPainter.java

```diff
@@ -201,7 +229,18 @@ private boolean createLink(final Object fromVertex, final Object toVertex, final
         if (toVertex instanceof ComponentBuilder) {
             final ComponentBuilder componentBuilder = (ComponentBuilder) toVertex;
             if (sourceColumns != null && !sourceColumns.isEmpty()) {
+
                 try {
+                    if (sourceAnalysisJobBuilder != componentBuilder.getAnalysisJobBuilder()){
+                        if(componentBuilder.getInput().length > 0) {
+                            logger.debug("createLink(...) returning false - cannot change analysisJobBuilder for a component with configured inputColumns");
+                            return false;
+                        }
+                        componentBuilder.getAnalysisJobBuilder().removeComponent(componentBuilder);
+                        sourceAnalysisJobBuilder.addComponent(componentBuilder);
+                        componentBuilder.setAnalysisJobBuilder(sourceAnalysisJobBuilder);
```

Comment:
Maybe avoid this "remove + add + set" sequence have a single method (in either AnalysisJobBuilder or ComponentBuilder) that "moves" the component to a different scope?